sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -51,16 +51,31 @@ import org.apache.spark.util.collection.OpenHashMap
       _FUNC_(col, array(percentage1 [, percentage2]...)) - Returns the exact percentile value array
       of numeric column `col` at the given percentage(s). Each value of the percentage array must
       be between 0.0 and 1.0.
-    """)
+
+      _FUNC_(col, frequency, percentage) - Returns the exact percentile value of numeric
+      column `col` with frequency column `frequency` at the given percentage. The value of
+      percentage must be between 0.0 and 1.0.
+
+      _FUNC_(col, frequency, array(percentage1 [, percentage2]...)) - Returns the exact percentile
+      value array of numeric column `col` with frequency column `frequency` at the given
+      percentage(s).Each value of the percentage array must be between 0.0 and 1.0.
+
+      """)
 case class Percentile(
     child: Expression,
+    frequency : Expression,
     percentageExpression: Expression,
+    withFrqExpr : Boolean,
```

Comment:
Let's make the type of `frequency` to be `Option[Expression]`, and remove the `withFrqExpr` flag.